src/overworld.c

```diff
@@ -400,30 +400,27 @@ void ApplyNewEncryptionKeyToGameStats(u32 newKey)
 
 // Routines related to object events
 
-static void sub_8054F68(void)
+static void LoadObjEventTemplatesFromHeader(void)
 {
     u8 i, j;
-    u8 mapGroup;
-    u8 mapNum;
-    u8 localId;
-    const struct MapHeader * linkedMap;
-
     for (i = 0, j = 0; i < gMapHeader.events->objectEventCount; i++)
     {
-        if (gMapHeader.events->objectEvents[i].unk2 == 0xFF)
+        if (gMapHeader.events->objectEvents[i].inConnection == 0xFF)
         {
-            localId = gMapHeader.events->objectEvents[i].elevation;
-            mapNum = gMapHeader.events->objectEvents[i].trainerType;
-            mapGroup = gMapHeader.events->objectEvents[i].trainerRange_berryTreeId;
-            linkedMap = Overworld_GetMapHeaderByGroupAndId(mapGroup, mapNum);
-            gSaveBlock1Ptr->objectEventTemplates[j] = linkedMap->events->objectEvents[localId - 1];
+            // load "in_connection" object from the connecting map
+            u8 localId = gMapHeader.events->objectEvents[i].elevation;
+            u8 mapNum = gMapHeader.events->objectEvents[i].trainerType;
+            u8 mapGroup = gMapHeader.events->objectEvents[i].trainerRange_berryTreeId;
```

Comment:
Are you sure these auto var names make sense?

Reply:
Yup. in_connection objects use these fields a little oddly, but it needs to know which map/object to replace it with during the transition